src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -652,6 +660,10 @@ private function addService($id, $definition)
         if ($definition->isSynthetic()) {
             $code .= sprintf("        throw new RuntimeException('You have requested a synthetic service (\"%s\"). The DIC does not know how to construct this service.');\n    }\n", $id);
         } else {
+            if ($definition->isDeprecated()) {
+                $code .= sprintf("        @trigger_error('%s', E_USER_DEPRECATED);\n\n", addslashes($definition->getDeprecationMessage($id)));
```

Comment:
you should use ``var_export`` instead to export the string, to account for more cases